Sources/_StringProcessing/Regex/DSLTree.swift

```diff
@@ -164,6 +164,96 @@ extension DSLTree {
       indirect case subtraction(CustomCharacterClass, CustomCharacterClass)
       indirect case symmetricDifference(CustomCharacterClass, CustomCharacterClass)
     }
+    
+    public struct AsciiBitset {
+      let isInverted: Bool
+      let isCaseInsensitive: Bool
+      var a: UInt64 = 0
+      var b: UInt64 = 0
+      
+      init(isInverted: Bool, isCaseInsensitive: Bool) {
+        self.isInverted = isInverted
+        self.isCaseInsensitive = isCaseInsensitive
+      }
+      
+      init(_ val: UInt8, _ isInverted: Bool, _ isCaseInsensitive: Bool) {
+        self.isInverted = isInverted
+        self.isCaseInsensitive = isCaseInsensitive
+        add(val)
+      }
+      
+      init(low: UInt8, high: UInt8, isInverted: Bool, isCaseInsensitive: Bool) {
+        self.isInverted = isInverted
+        self.isCaseInsensitive = isCaseInsensitive
+        for val in low...high {
+          add(val)
+        }
+      }
+      
+      internal init(
+        a: UInt64,
+        b: UInt64,
+        isInverted: Bool,
+        isCaseInsensitive: Bool
+      ) {
+        self.isInverted = isInverted
+        self.isCaseInsensitive = isCaseInsensitive
+        self.a = a
+        self.b = b
+      }
+      
+      internal mutating func add(_ val: UInt8) {
+        setBit(val)
+        if isCaseInsensitive {
+          let c = Character(Unicode.Scalar.init(val))
+          let otherCase: String
+          if c.isUppercase {
+            otherCase = c.lowercased()
+          } else {
+            otherCase = c.uppercased()
+          }
+          setBit(otherCase.first!.asciiValue!)
```

Comment:
I suppose this works, but ideally we'd have fewer assumptions baked in. @Azoy would just asking for the scalar's mapping work too?

Reply:
Alternatively, since it's pure ASCII, we could just compute and apply the mapping.

Reply:
If we're only dealing with ASCII here, we should probably just compute the uppercase/lowercase since there's relatively few scalars in ASCII that actually have uppercase/lowercase (is it just the letters?)